behaviour/graql/language/match.feature

```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has name $y; get $x;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | LEI | key   | ref:0 |
+      | GRA | key   | ref:2 |
+    Then uniquely identify answer concepts
+      | x   |
+      | LEI |
+      | GRA |
+
+
+  Scenario: by using the `attribute` meta label, `has` can be used to match things that own any type of attribute with a specified value
+    Given graql define
+      """
+      define
+      shoe-size sub attribute, value long;
+      person has shoe-size;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has age 9, has ref 0;
+      $y isa person, has shoe-size 9, has ref 1;
+      $z isa person, has age 12, has shoe-size 12, has ref 2;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has attribute 9; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | AG9 | key   | ref:0 |
+      | SS9 | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | AG9 |
+      | SS9 |
+
+
+  Scenario: when an attribute instance is fully specified, `has` matches its owners
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "John", has ref 0;
+      $y isa person, has name "John", has ref 1;
+      $z isa person, has name "Not John", has ref 2;
+      $w isa person, has ref 3;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has name "John"; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | JH1 | key   | ref:0 |
+      | JH2 | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | JH1 |
+      | JH2 |
+
+
+  Scenario: `has` with an instance fully specified matches all its owners, even if they own other instances of the same attribute
+    Given graql define
+      """
+      define
+      lucky-number sub attribute, value long;
+      person has lucky-number;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has lucky-number 10, has lucky-number 20, has lucky-number 30, has ref 0;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has lucky-number 20; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | PER | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | PER |
+
+
+  ##############################
+  # ATTRIBUTE VALUE COMPARISON #
+  ##############################
+
+  Scenario: when two things each own an attribute with the same value, and those attributes have different types, they will match by equality, but not by ownership
```

Comment:
same here

Reply:
`when things own attributes of different types but the same value, they match by equality, but not ownership`